acceptance/tests/provider/package/pip.rb

```diff
@@ -22,21 +26,42 @@
       end
     end
 
-    step "Install a pip package" do
+    step "Ensure presence of a pip package" do
       package_manifest = resource_manifest('package', package, { ensure: 'present', provider: 'pip' } )
       apply_manifest_on(agent, package_manifest, :catch_failures => true) do
         list = on(agent, "#{pip_command} list --disable-pip-version-check").stdout
         assert_match(/#{package} \(/, list)
       end
-      on(agent, "#{pip_command} uninstall #{package} --disable-pip-version-check --yes")
+    end
+
+    step "Install a pip package using version range" do
+      package_manifest1 = resource_manifest('package', package, { ensure: '<=1.1.0', provider: 'pip' } )
+      package_manifest2 = resource_manifest('package', package, { ensure: '<1.0.4',  provider: 'pip' } )
+
+      # Ensure the package is not already installed
+      on(agent, "#{pip_command} uninstall #{package} --disable-pip-version-check --yes", :accept_all_exit_codes => true)
+
+      # Make a fresh package install (with version lower than or equal to 1.1.0)
+      apply_manifest_on(agent, package_manifest1, :expect_changes => true) do
+        list = on(agent, "#{pip_command} list --disable-pip-version-check").stdout
+        assert_match(/#{package} \(/, list)
```

Comment:
Here you could also assert in version installed

Reply:
Asserted